app/lib/shared/redis_cache.dart

```diff
@@ -223,3 +225,57 @@ Future _withInmemoryCache(FutureOr Function() fn) async {
   _registerCache(CachePatterns._(Cache(Cache.inMemoryCacheProvider(4096))));
   return await fn();
 }
+
+typedef CacheProviderFn<T> = Future<CacheProvider<T>> Function();
+
+class _UpdatingCacheProvider<T> implements CacheProvider<T> {
```

Comment:
Can we name this better? `_PeriodicallyReconnectingCacheProvider` or `ConnectionRefreshingCacheProvider`

Reply:
Also a one or two line documentation comment wouldn't hurt...